Duplicati.Library.Backend.Tardigrade/TardigradeBackend.cs

```diff
@@ -0,0 +1,330 @@
+            { "API key", "API key" },
+            { "Access grant", "Access grant" },
+        };
+
+        // ReSharper disable once UnusedMember.Global
+        // This constructor is needed by the BackendLoader.
+        public Tardigrade()
+        {
+        }
+
+        // ReSharper disable once UnusedMember.Global
+        // This constructor is needed by the BackendLoader.
+        public Tardigrade(string url, Dictionary<string, string> options)
+        {
+            var auth_method = options[TARDIGRADE_AUTH_METHOD];
+
+            Access.SetTempDirectory(System.IO.Path.GetTempPath());
+
+            if (auth_method == "Access grant")
+            {
+                //Create an access from the access grant
+                var shared_access = options[TARDIGRADE_SHARED_ACCESS];
+                _access = new Access(shared_access);
+            }
+            else
+            {
+                //Create an access for a satellite, API key and encryption passphrase
+                _satellite = options[TARDIGRADE_SATELLITE];
+
+                if (options.ContainsKey(TARDIGRADE_API_KEY))
+                {
+                    _api_key = options[TARDIGRADE_API_KEY];
+                }
+                if (options.ContainsKey(TARDIGRADE_SECRET))
+                {
+                    _secret = options[TARDIGRADE_SECRET];
+                }
+
+                _access = new Access(_satellite, _api_key, _secret);
+            }
+
+            _bucketService = new BucketService(_access);
+            _objectService = new ObjectService(_access);
+
+            //If no bucket was provided use the default "duplicati"-bucket
+            if (options.ContainsKey(TARDIGRADE_BUCKET))
+            {
+                _bucket = options[TARDIGRADE_BUCKET];
+            }
+            else
+            {
+                _bucket = "duplicati";
+            }
+
+            if (options.ContainsKey(TARDIGRADE_FOLDER))
+            {
+                _folder = options[TARDIGRADE_FOLDER];
+            }
+        }
+
+        public string DisplayName
+        {
+            get { return Strings.Tardigrade.DisplayName; }
+        }
+
+        public string ProtocolKey => PROTOCOL_KEY;
+
+        public IList<ICommandLineArgument> SupportedCommands
+        {
+            get
+            {
+                return new List<ICommandLineArgument>(new ICommandLineArgument[] {
+                    new CommandLineArgument(TARDIGRADE_AUTH_METHOD, CommandLineArgument.ArgumentType.String, Strings.Tardigrade.TardigradeAuthMethodDescriptionShort, Strings.Tardigrade.TardigradeAuthMethodDescriptionLong, "API key"),
+                    new CommandLineArgument(TARDIGRADE_SATELLITE, CommandLineArgument.ArgumentType.String, Strings.Tardigrade.TardigradeSatelliteDescriptionShort, Strings.Tardigrade.TardigradeSatelliteDescriptionShort, "us-central-1.tardigrade.io:7777"),
```

Comment:
Should the 4th argument be `Strings.Tardigrade.TardigradeSatelliteDescriptionLong`?

Reply:
True, Thanks. Resolved!